### ISSUE TITLE
This breaks compatibility with small repositories where there is file renaming

### ISSUE THREAD

**Issue by lunny** [go-gitea/gitea#22912]
This breaks compatibility with small repositories where there is file renaming

_Originally posted by @bvp in https://github.com/go-gitea/gitea/issues/20764#issuecomment-1430987108_

@bvp

**Comment by moseslecce** [comment]
Wouldn't it be best to offer this as a setting, perhaps per repo, where the administrator can decide if there is value to have this flag on or off (to decide if they want performance or a full historical view of changes)?

Alternatively [github offers a link to show extended history](https://github.blog/changelog/2022-06-06-view-commit-history-across-file-renames-and-moves/)

**Comment by martin-tag24** [comment]
> Wouldn't it be best to offer this as a setting, perhaps per repo, where the administrator can decide if there is value to have this flag on or off (to decide if they want performance or a full historical view of changes)?
> 
> Alternatively [github offers a link to show extended history](https://github.blog/changelog/2022-06-06-view-commit-history-across-file-renames-and-moves/)

Yes please make `--follow` as repo setting or add it as option to the api and commits list.

**Comment by bircni** [comment]
This should be addressed by #34994 ("feat(ui): add 'follow rename' to file commit history list"), merged and slated for the 1.27.0 release. After #20764 dropped `git log --follow` for performance, that PR re-introduces rename-following as an explicit option in the file commit history view, so repositories with file renames can once again view the full history. Closing as resolved — please reopen if the new option doesn't cover your use case.
